bids/reports/utils.py

```diff
@@ -0,0 +1,573 @@
+
+    if 'IntendedFor' in metadata.keys():
+        scans = metadata['IntendedFor']
+        run_dict = {}
+        for scan in scans:
+            fn = basename(scan)
+            run_search = re.search(r'.*_run-([0-9]+).*', fn)
+            run_num = int(run_search.groups()[0])
+            type_search = re.search(r'.*_([a-z0-9]+)\..*', fn)
+            ty = type_search.groups()[0].upper()
+            if ty == 'BOLD':
+                task_search = re.search(r'.*_task-([a-z0-9]+).*', fn)
+                task = config['task'].get(task_search.groups()[0],
+                                          task_search.groups()[0])
+                ty_str = '{0} {1} scan'.format(task, ty)
+            else:
+                ty_str = '{0} scan'.format(ty)
+
+            if ty_str not in run_dict.keys():
+                run_dict[ty_str] = []
+            run_dict[ty_str].append(run_num)
+
+        for scan in run_dict.keys():
+            run_dict[scan] = [num2words(r, ordinal=True) for r in sorted(run_dict[scan])]
+
+        out_list = []
+        for scan in run_dict.keys():
+            if len(run_dict[scan]) > 1:
+                s = 's'
+            else:
+                s = ''
+            run_str = list_to_str(run_dict[scan])
+            string = '{rs} run{s} of the {sc}'.format(rs=run_str,
+                                                      s=s,
+                                                      sc=scan)
+            out_list.append(string)
+        for_str = ' for the {0}'.format(list_to_str(out_list))
+    else:
+        for_str = ''
+
+    desc = '''
+           A {variants} {seqs} field map (phase encoding:
+           {dir_}; {n_slices} slices; repetition time, TR={tr}ms;
+           echo time, TE={te}ms; flip angle, FA={fa}<deg>;
+           field of view, FOV={fov}mm; matrix size={ms};
+           voxel size={vs}mm) was acquired{for_str}.
+           '''.format(variants=variants,
+                      seqs=seqs,
+                      dir_=dir_,
+                      for_str=for_str,
+                      n_slices=n_slices,
+                      tr=num_to_str(metadata['RepetitionTime']*1000),
+                      te=num_to_str(metadata['EchoTime']*1000),
+                      fa=metadata['FlipAngle'],
+                      vs=vs_str,
+                      fov=fov_str,
+                      ms=ms_str)
+    desc = desc.replace('\n', ' ').lstrip()
+    while '  ' in desc:
+        desc = desc.replace('  ', ' ')
+
+    return desc
+
+
+def final_paragraph(metadata):
+    """
+    Describes dicom-to-nifti conversion process and methods generation.
+
+    Parameters
+    ----------
+    metadata : :obj:`dict`
+        The metadata for the scan.
+
+    Returns
+    -------
+    desc : :obj:`str`
+        Output string with scanner information.
+    """
+    if 'ConversionSoftware' in metadata.keys():
+        software_str = ' using {soft} ({conv_vers})'.format(soft=metadata['ConversionSoftware'],
```

Comment:
Indentation looks off (though it could just be a GitHub issue); maybe run this all through a linter if you haven't already, and fix any glaring PEP8 issues.